src/api/c/moments.cpp

```diff
@@ -29,14 +29,14 @@
 using af::dim4;
 using namespace detail;
 
-template<typename T>
+template<typename T, typename To>
 static inline void moments(af_array *out, const af_array in, af_moment_type moment)
 {
     Array<float> temp = moments<T>(getArray<T>(in), moment);
     af_array tarr = getHandle<float>(temp);
-    Array<T> output = castArray<T>(tarr);
+    Array<To> output = castArray<To>(tarr);
 
-    *out = getHandle<T>(output);
+    *out = getHandle<To>(output);
```

Comment:
@syurkevi I think the output should always be float. Is there any reason to cast it to `To` ? Casting it to `To` (or `T` for that matter) results in incorrect values for higher order moments.

Reply:
Leaving everything as floats caused tests to fail for ints and doubles. This is because the tests expect the same output type as input type. I changed the tests, let me know if it's alright or if I should revert.